apps/web/src/components/forms/signin.tsx

```diff
@@ -122,6 +130,15 @@ export const SignInForm = ({ className }: SignInFormProps) => {
 
         const errorMessage = ERROR_MESSAGES[result.error];
 
+        if (result.error === ErrorCode.UNVERIFIED_EMAIL) {
+          const user = await getUser({ email });
+          const token = user?.VerificationToken[user.VerificationToken.length - 1].token;
```

Comment:
Doing this workaround to avoid exposing the email address in the URL. The assumption is that a newly-registered user will always have at least 1 verification token. We'll use the last verification token and identify the user on the `/unverified-account` page, whenever the user clicks the "resend confirmation email" button.

If you can think of a better solutions, it is more than welcome!

Reply:
This is going to cause an issue because you're exposing the token to the user.

For example, the user signs in with an unconfirmed email and gets redirected to:

http://localhost:3000/unverified-account?t=72cc56a3c55dbb5a3c60e3d8203f6c20f345fb2b

They can take the token in the search param, and go directly to `/verify-email` and have the email be verified

http://localhost:3000/verify-email/72cc56a3c55dbb5a3c60e3d8203f6c20f345fb2b

A better solution would be to encrypt the email into the search param and decrypt it

There's a pending PR for this now (if you could please review it if you have time): 

https://github.com/documenso/documenso/pull/863